src/tests/suite/daos_obj.c

```diff
@@ -2287,8 +2288,11 @@ basic_byte_array(void **state)
 	assert_memory_equal(buf, buf_out, buf_len);
 
 	print_message("short read should get iov_len with tail hole trimmed\n");
-	memset(buf_out, 0, buf_len);
 	tmp_len = buf_len / 3;
+	buf_out_tmp = buf_out;
+	D_ALLOC(buf_out, max(buf_len, tmp_len + 99));
+	D_ASSERT(buf_out != NULL);
+	memset(buf_out, 0, buf_len);
```

Comment:
NIT, not sure it is needed as  D_ALLOC is calling calloc and thus the memory is already set to zero.

Reply:
Indeed. Let me remove this `memset` call.